### PR TITLE
Add MySQL OTel input package

### PR DESCRIPTION
**Pull request by perk**
## Proposed commit message

Add MySQL OpenTelemetry input package (`mysql_input_otel`).

**WHAT:** New input package that enables collection of MySQL metrics using the OpenTelemetry Collector's [mysqlreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver). The package provides full configuration support for the receiver: connection settings (endpoint, username, password, database, transport, TLS), collection settings (interval, initial delay), statement events, query sample collection, and top query collection. Includes Docker test environment (MySQL 9.6), policy and system tests, input template (`input.yml.hbs`), manifest, docs, changelog, and CODEOWNERS entry for `@elastic/ecosystem`. Works with the existing `mysql_otel` assets package for dashboards and visualizations.

**WHY:** To allow users to collect MySQL metrics via the Elastic Agent with the EDOT Collector ~(9.1.x+)~ (^9.2.2) using the standard OpenTelemetry MySQL receiver, with a supported input package and integration with the existing MySQL OTel assets.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] ~I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)~

## Author's Checklist

- [x] Input template and manifest options match upstream mysqlreceiver behavior
- [x] Policy and system test expectations are correct for default and TLS variants
- [x] README and configuration table are accurate

## How to test this PR locally

1. From the repo root, go to the package: `cd packages/mysql_input_otel`
2. Start the Docker test environment: `elastic-package stack up -d --services mysql_input_otel` (or use `_dev/deploy/docker/docker-compose.yml` with `docker compose up -d`)
3. Run policy tests: `elastic-package test policy`
4. Run system tests: `elastic-package test system`
5. To test end-to-end: add the MySQL OTel input to an agent policy (endpoint `localhost:3306`, username `root`, no password when using the dev MySQL container), deploy the agent, and confirm metrics in Kibana with the `mysql_otel` assets package

## Related issues

- Resolves https://github.com/elastic/integrations/issues/17144

## Screenshots

Screenshot 2026-02-10 at 12 01 50

**Related:**
- MySQL receiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver
- MySQL OpenTelemetry Assets Package: [mysql_otel](https://www.elastic.co/docs/reference/integrations/mysql_otel)

### CODE REVIEW

**Comment by strawgate** [comment]
@claude please do a thorough review -- pay careful attention to the receiver upstream and at our other input packages

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -346,6 +346,7 @@
 /packages/mysql @elastic/obs-infraobs-integrations
 /packages/mysql_enterprise @elastic/sec-windows-platform
 /packages/mysql_otel @elastic/obs-infraobs-integrations
+/packages/mysql_otel_input @elastic/ecosystem
```
Comment (mrodm) [inline .github/CODEOWNERS]: Please rename this package to be `mysql_input_otel` to be consistent with the other input packages.

A part from changing the package folder, it would require to change the package name in the manifest as well as maybe the expected files for tests (and sample_event.json).

File: packages/mysql_input_otel/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+|---------|----------|---------|-------------|
+| Lookback Time | No | `60` | Time window (seconds) for top queries |
+| Max Query Sample Count | No | `1000` | Maximum records to fetch per run |
+| Top Query Count | No | `200` | Maximum active queries to report |
+| Collection Interval | No | `60s` | Interval for top query emission |
+| Query Plan Cache Size | No | `1000` | Cache size for query plan results |
+| Query Plan Cache TTL | No | `1h` | TTL for cached query plans |
+
+
+## Metrics reference
```
Comment (mrodm) [inline packages/mysql_input_otel/docs/README.md:21]: Please add a new section for the configuration optoins to point to the upstream configuration. Something like:

```
## Configuration

For the full list of settings exposed for the receiver and examples, refer to the [configuration](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/mysqlreceiver#configuration) section.

```

File: packages/mysql_input_otel/_dev/test/system/test-default-config.yml
Comment (mrodm) [inline packages/mysql_input_otel/_dev/test/system/test-default-config.yml:1]: As there are system tests defined, please add a `sample_event.json` file so it also runs static tests.
This file can be generated via
```bash
elastic-package test system -v --generate
```

File: packages/mysql_otel_input/agent/input/input.yml.hbs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+{{else if tls_insecure_skip_verify}}
+    tls:
+      insecure_skip_verify: {{tls_insecure_skip_verify}}
+{{#if tls_server_name_override}}
+      server_name_override: {{tls_server_name_override}}
+{{/if}}
+{{else if tls_server_name_override}}
+    tls:
+      server_name_override: {{tls_server_name_override}}
+{{/if}}
```
Comment (mrodm) [inline packages/mysql_otel_input/agent/input/input.yml.hbs]: This way of setting tls variables looks complicated, I don't know if tehre could be some edge cases with wrong parameters. But not sure if this could be simplified. O

File: packages/mysql_input_otel/_dev/test/policy/test-default.expected
Comment (mrodm) [inline packages/mysql_input_otel/_dev/test/policy/test-default.expected:1]: Please add some tests to validate that TLS settings are rendered as expected 🙏

File: packages/mysql_otel_input/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        title: Username
+        description: The username used to connect to MySQL.
+        default: root
+        show_user: true
+      - name: password
+        type: password
+        required: false
+        title: Password
+        description: The password used to connect to MySQL.
+        secret: true
```
[the remark this answers is not in the document]
Comment (mrodm) [reply packages/mysql_otel_input/manifest.yml]: In redis PR I've added this TODO for this:

```yaml
        # TODO: Change to secret: true once Fleet Server resolves the secret handling issue
        # for OTel input packages. See: https://github.com/elastic/fleet-server/issues/6277
        secret: false
```

File: packages/mysql_input_otel/_dev/test/system/test-tls-config.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+service: mysql_input_otel
+vars:
+  endpoint: "{{Hostname}}:{{Port}}"
+  username: "root"
+  tls_enabled: true
```
[the remark this answers is not in the document]
Comment (teresaromero) [reply packages/mysql_input_otel/_dev/test/system/test-tls-config.yml]: removed system test

File: packages/mysql_input_otel/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+      - name: password
+        type: password
+        required: false
+        title: Password
+        description: The password used to connect to MySQL.
+        # TODO: Change to secret: true once Fleet Server resolves the secret handling issue
+        # for OTel input packages. See: https://github.com/elastic/fleet-server/issues/6277
+        secret: false
+        show_user: true
+      - name: database
```
[the remark this answers is not in the document]
Comment (teresaromero) [reply packages/mysql_input_otel/manifest.yml:55]: `allow_native_passwords` fixed

File: packages/mysql_input_otel/_dev/test/policy/test-default.expected [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        collection_interval: 10s
+        endpoint: localhost:3306
+        initial_delay: 1s
+        query_sample_collection:
+            max_rows_per_query: 100
+        statement_events:
+            digest_text_limit: 120
+            limit: 250
+            time_limit: 24h
+        top_query_collection:
```
Comment (teresaromero) [inline packages/mysql_input_otel/_dev/test/policy/test-default.expected:45]: this property is supported since mysqlreceiver [v0.138.0](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mysqlreceiver@v0.138.0#readme-configuration)

this is introduced at the elastic-stack since v.9.2.2, so i've updated the constraint for the input package to be compatible since v9.2.2

File: packages/mysql_input_otel/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+  license: "Elastic-2.0"
+description: "Collect MySQL metrics using OpenTelemetry Collector"
+type: input
+categories:
+  - datastore
+  - observability
+  - opentelemetry
+conditions:
+  kibana:
+    version: "^9.2.2"
```
Comment (teresaromero) [inline packages/mysql_input_otel/manifest.yml:15]: minimum version of the stack supporting all the fields for configuring the receiver, related to https://github.com/elastic/integrations/pull/17090/changes#r2792654798

File: packages/mysql_input_otel/_dev/deploy/docker/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+services:
+  mysql_input_otel:
+    image: mysql:${SERVICE_VERSION:-9.6.0}
```
Comment (mrodm) [inline packages/mysql_input_otel/_dev/deploy/docker/docker-compose.yml]: As it has been removed v9 from variants, I would also change this default:

```suggestion
    image: mysql:${SERVICE_VERSION:-8.4.8}
```

File: packages/mysql_input_otel/_dev/build/docs/README.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+### Top Query Collection Settings
+
+| Setting | Required | Default | Description |
+| --------- | ---------- | --------- | ------------- |
+| Lookback Time | No | `60` | Time window (seconds) for top queries |
+| Max Query Sample Count | No | `1000` | Maximum records to fetch per run |
+| Top Query Count | No | `200` | Maximum active queries to report |
+| Collection Interval | No | `60s` | Interval for top query emission |
+| Query Plan Cache Size | No | `1000` | Cache size for query plan results |
+| Query Plan Cache TTL | No | `1h` | TTL for cached query plans |
```
Comment (mrodm) [inline packages/mysql_input_otel/_dev/build/docs/README.md]: As there is a section (Configuration) pointing to the upstream configuration, I think this one could be removed. It could be outdated easily.

File: packages/mysql_input_otel/_dev/deploy/docker/docker-compose.yml (hunk 9 above)
Comment (teresaromero) [reply packages/mysql_input_otel/_dev/deploy/docker/docker-compose.yml]: thnks

File: packages/mysql_input_otel/_dev/test/system/test-default-config.yml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+service: mysql_input_otel
+vars:
+  endpoint: "{{Hostname}}:{{Port}}"
+  username: "root"
```
Comment (mrodm) [inline packages/mysql_input_otel/_dev/test/system/test-default-config.yml:4]: Nit.
It could be added these settings to ensure that a minimum number of documents are ingested into Elasticsearch

```yaml
assert:
  min_count: 50
```

**Comment by teresaromero** [comment]
pinging @elastic/obs-infraobs-integrations as codeowers for `mysql` package to have a look into this implementation
